### PR TITLE
bug: Improve quotation pattern.

### PR DESCRIPTION
**Pull request by michael-simons**
* Too many escapes of the backslash
* No need to check for the actual ` and the unicode value
* BUT a need to check for the escaped unicode proper (which is literael
  \\u0060, written down as Java String becomes \\\\u0060, no further
  escapeing for the regex necessary

### CODE REVIEW

**Comment by michael-simons** [comment]
Paging @AlexNeo4J

**Comment by michael-simons** [comment]
Hey @AzuObs @gem-neo4j @AlexNeo4J @nadja-muller I have upped the escaping, sanitising and quotation game a bit. With the `SchemaNames` class in the module that is added in this commit I am able to provide safe quotation and sanitising of (Java/JVM) based input across a broad range of version, see the integration test in the same module. 

The goal was have excellent behaviour on 4.4 by default, and - if a caller has those information available - the same behaviour on older versions.

WIth the module it is possible to achieve 5.0 (or 6.0, depending on when CIP-59 is implemented), today. 

Let me know if I missed something in my dissy head last night.

Also, @nadja-muller: In the CIP you speak about supporting UTF-32 literals, but those will contradict the nice example with multiple resolution of unicode literals, respectively made it rather hard (how to decide if `\u1234` or `\u123456` should be resolved if both are valid codepoints, if you go for the longest possible, than the example from the CIP will (the `something…`one).

File: neo4j-cypher-dsl-schema-name-support/src/test/java/org/neo4j/cypherdsl/support/schema_name/SchemaNamesIT.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		// Already escaped backslash
+		{ "x\\\\y", "x\\y" },
+		// Escaped backticks
+		{ "x\\`y", "x`y" },
+		{ "x\\```y", "x``y" },
+		{ "x`\\```y", "x```y" },
+		// This is the backtick itself in the string
+		{ "Foo \u0060", "Foo `" },
+		// This is the backtick unicode escaped so that without further processing `foo \u0060` would end up at Cypher,
+		{ "Foo \\u0060", "Foo `" },
```
Comment (AzuObs) [inline neo4j-cypher-dsl-schema-name-support/src/test/java/org/neo4j/cypherdsl/support/schema_name/SchemaNamesIT.java:98]: I didn't check all the source code, but I mainly looked at the tests.

I think these would work but is it worth trying some combinations involved unicode for backslash?
eg ```"\u005Du0060", "`"```

Comment (michael-simons) [reply neo4j-cypher-dsl-schema-name-support/src/test/java/org/neo4j/cypherdsl/support/schema_name/SchemaNamesIT.java:105]: Excellent catch.

**Comment by michael-simons** [comment]
I categorized the tests a bit

**Pull request merged**